src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala

```diff
@@ -140,4 +141,39 @@ object EthJsonMethodsImplicits extends JsonMethodsImplicits {
 
     def encodeJson(t: SendRawTransactionResponse): JValue = encodeAsHex(t.transactionHash)
   }
+
+  implicit val eth_call = new JsonDecoder[CallRequest] with JsonEncoder[CallResponse] {
+    def decodeJson(params: Option[JArray]): Either[JsonRpcError, CallRequest] =
+      params match {
+        case Some(JArray((txObj: JObject) :: (blockStr: JString) :: Nil)) =>
+          for {
+            blockParam <- tryExtractBlockParam(blockStr)
+            tx <- extractTx(txObj)
+          } yield CallRequest(tx, blockParam)
+        case _ => Left(InvalidParams())
+      }
+
+    def encodeJson(t: CallResponse): JValue = encodeAsHex(t.returnData)
+
+    def extractTx(obj: JObject): Either[JsonRpcError, CallTx] = {
+      def toEitherOpt[A, B](opt: Option[Either[A, B]]): Either[A, Option[B]] =
+        opt.map(_.right.map(Some.apply)).getOrElse(Right(None))
+
+      for {
+        from <- toEitherOpt((obj \ "from").extractOpt[String].map(tryExtractUnformattedData))
```

Comment:
What does `(obj \ "from").extractOpt[String]` return if the `from` not a string (eg. `{"from": 42}`) ?